diff -uNdr motion-Release-4.3.0_orig/src/netcam.h motion-Release-4.3.0_new/src/netcam.h
--- motion-Release-4.3.0_orig/src/netcam.h	2020-02-05 09:28:50.329199353 +0300
+++ motion-Release-4.3.0_new/src/netcam.h	2020-02-05 09:30:52.591677971 +0300
@@ -113,7 +113,7 @@
 } netcam_buff;
 typedef netcam_buff *netcam_buff_ptr;
 
-struct netcam_caps {                    /* netcam capabilities: */
+extern struct netcam_caps {                    /* netcam capabilities: */
         unsigned char streaming;        /*  See the NCS_* defines */
         unsigned char content_length;   /*  0 - unsupported     */
 } caps;
diff -uNdr motion-Release-4.3.0_orig/src/translate.c motion-Release-4.3.0_new/src/translate.c
--- motion-Release-4.3.0_orig/src/translate.c	2020-02-05 09:28:50.346198723 +0300
+++ motion-Release-4.3.0_new/src/translate.c	2020-02-05 09:40:22.322733174 +0300
@@ -10,6 +10,8 @@
 #include "motion.h"
 #include "translate.h"
 
+int nls_enabled;
+
 void translate_locale_chg(const char *langcd){
 #ifdef HAVE_GETTEXT
     /* This routine is for development testing only.  It is not used for
diff -uNdr motion-Release-4.3.0_orig/src/translate.h motion-Release-4.3.0_new/src/translate.h
--- motion-Release-4.3.0_orig/src/translate.h	2020-02-05 09:28:50.346198723 +0300
+++ motion-Release-4.3.0_new/src/translate.h	2020-02-05 09:40:22.313733505 +0300
@@ -10,7 +10,7 @@
 #ifndef _INCLUDE_TRANSLATE_H_
 #define _INCLUDE_TRANSLATE_H_
 
-int nls_enabled;
+extern int nls_enabled;
 
 #ifdef HAVE_GETTEXT
 #   include <libintl.h>
